addons/mail/static/src/core/common/message.xml

```diff
@@ -148,15 +148,27 @@
             }"
         >
             <t t-foreach="messageActions.actions.slice(0, quickActionCount)" t-as="action" t-key="action.id">
+                <t t-set="firstAction" t-value="action.id === messageActions.actions[0].id"/>
+                <t t-set="lastAction" t-value="messageActions.actions.length gt quickActionCount and action.id === messageActions.actions[messageActions.actions.length - 1].id"/>
                 <t t-if="action.callComponent" t-component="action.callComponent" t-props="action.props"/>
-                <button t-else="" class="btn px-1 py-0 rounded-0" t-att-title="action.title" t-att-name="action.id" t-on-click.stop="action.onClick">
+                <button
+                    t-else=""
+                    class="btn px-1 py-0 rounded-0"
+                    t-att-class="{
+                        'rounded-start-1': firstAction,
+                        'rounded-end-1': lastAction,
+                    }"
+                    t-att-title="action.title"
+                    t-att-name="action.id"
+                    t-on-click.stop="action.onClick"
+                >
                     <i class="fa fa-lg" t-att-class="action.icon"/>
                 </button>
             </t>
             <div t-if="messageActions.actions.length gt quickActionCount" class="d-flex rounded-0">
                 <Dropdown state="optionsDropdown" position="props.message.threadAsNewest  ? 'top-start' : 'bottom-start'" menuClass="'d-flex flex-column py-0 o-mail-Message-moreMenu'" >
-                    <button t-att-class="`btn p-0 rounded-0 ${ optionsDropdown.isOpen ? 'bg-200' : '' }`" t-att-title="expandText">
-                        <i class="btn px-1 py-0 fa fa-lg fa-ellipsis-h rounded-0" t-att-class="{ 'order-1': props.isInChatWindow }" tabindex="1"/>
+                    <button t-att-class="`btn px-1 py-0 rounded-0 rounded-end-1 ${ optionsDropdown.isOpen ? 'bg-200' : '' }`" t-att-title="expandText">
```

Comment:
Again this is not necessarily the right-most item. e.g. in chat window self-messages are aligned right, so expand is far-left quick action.

This should also be a conditional classname.